core/state_processor.go

```diff
@@ -234,14 +235,24 @@ func ProcessBeaconBlockRoot(beaconRoot common.Hash, evm *vm.EVM) {
 }
 
 // ProcessParentBlockHash stores the parent block hash in the history storage contract
-// as per EIP-2935.
-func ProcessParentBlockHash(prevHash common.Hash, evm *vm.EVM) {
+// as per EIP-2935/7709.
+func ProcessParentBlockHash(prevHash common.Hash, evm *vm.EVM, is7709 bool) {
 	if tracer := evm.Config.Tracer; tracer != nil {
 		onSystemCallStart(tracer, evm.GetVMContext())
 		if tracer.OnSystemCallEnd != nil {
 			defer tracer.OnSystemCallEnd()
 		}
 	}
+	if is7709 {
+		// currently 8192 for kautinen7, to be changed to 8 (and made a constant) for kaustinen8
+		ringIndex := (evm.Context.BlockNumber.Uint64() - 1) % params.HistoryServeWindow
+		var key common.Hash
+		binary.BigEndian.PutUint64(key[24:], ringIndex)
+		evm.StateDB.SetState(params.SystemAddress, key, prevHash)
+		evm.StateDB.AccessEvents().SlotGas(params.SystemAddress, key, true)
+		evm.StateDB.Finalise(false)
```

Comment:
I am wondering if it's necessary. 

If Verkle is enabled, SStore will be replaced with `SStore4762`, the slot gas charging and warming will be done automatically if we register the block hash via invoking the contract, right?

Reply:
I guess this doesn't actually invoke any contract, just does the ugly and sets the slot. Which necessitates also setting the access-event.

Reply:
so you might have missed the gist of conversations I had over this topic. There are several issues with invoking a contract:

 1. it's slower and uglier as you have to instantiate an evm each time, and copy/paste the code to do so all over the place
 2. it's more brittle, since it can revert
 3. it's not extensible, if you need to change the behavior, you need to 
 4. you must filter the keys that bloat the witness a posteriori

Reply:
The conclusion was that, we would deprecate calling system contracts in verkle.